README.md

```diff
@@ -127,7 +127,8 @@ time_zone: null
 | animated_icon         | boolean          | **Optional** | Whether the large weather icon should be animated                                                                                                                                                                                 | `true`    |
 | forecast_rows         | number           | **Optional** | The amount of weather forecast rows to show. Depending on `hourly_forecast` each row either corresponds to a day or an hour                                                                                                       | `5`       |
 | locale                | string[^2]       | **Optional** | Language to use for language specific text and date/time formatting. If not provided, falls back to the locale set in HA or, if not set in HA, to `en-GB`                                                                         | `en-GB`   |
-| time_format           | `24` \| `12`     | **Optional** | Format used to display the time. If not provided, falls back to the default time format of the configured `locale`                                                                                                                | `24`      |
+| time_format           | `24` \| `12`     | **Optional** | Format used to display the time. If not provided, falls through to time_pattern format                                                                                                                                            | `null`    |
+| time_pattern          | string           | **Optional** | Pattern to use for time formatting. If not provided, falls back to a localized default time formatting. See [luxon](https://moment.github.io/luxon/#/formatting?id=table-of-tokens) for valid tokens                              | `HH:MM`   |
```

Comment:
precedence `time_pattern` -> `time_format` -> localized default time formatting.
default value should be `null`.

please also add the newly introduced property in `### Full configuration`